### PR TITLE
Add MacOS template

### PR DESCRIPTION
**Pull request by KaiserOfNone**
What it says on the tin.

### CODE REVIEW

**Review by Spartan322 — CHANGES_REQUESTED** [review]

File: game/export_presets.cfg [hunk 1]
```diff
@@ -106,3 +106,108 @@ Unregister-ScheduledTask -TaskName godot_remote_debug -Confirm:$false -ErrorActi
...
+
+name="macOS"
+platform="macOS"
+runnable=true
+dedicated_server=false
+custom_features=""
+export_filter="all_resources"
+include_filter=""
+exclude_filter=""
+export_path="export/osx/OpenVic2.dmg"
```
[the remark this answers is not in the document]
Comment (BrickPi) [reply game/export_presets.cfg:120]: Not necessarily, OSX is interchangeable with macOS and shorter, if anything clearer and just as consistent, coming from a mac user & developer.

[the remark this answers is not in the document]
Comment (Spartan322) [reply game/export_presets.cfg:120]: Well the standard right now is already present in the export directory, and its full descriptive.

**Comment by Spartan322** [comment]
One thing I suggest we do here is fix the clang mac compile issue before merge.

**Comment by Spartan322** [comment]
The compile issue seems to be caused by the use of `size_t` in Simulation.hpp, as `size_t` evaluates to an `unsigned long` however in godot-cpp, the exposed type relies upon `uint64_t` which if `__WORDSIZE` is 64 is the case, however if `__WORDSIZE` evaluates to 32, then `uint64_t` becomes `unsigned long long` instead.

**Comment by Spartan322** [comment]
Oops, wrongfully closed.

**Pull request merged**